### PR TITLE
Feature/ukr 217 update url component

### PR DESCRIPTION
**Pull request by dannewnsjump24**
This PR brings in a single vue component for the URL slug checker.  This also adds the relevant tests for the component and is a basis for us to start adding new compontents to the system

### CODE REVIEW

File: webpack.prod.js [hunk 1]
```diff
@@ -2,10 +2,13 @@ const path = require('path');
...
 const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
 const TerserPlugin = require('terser-webpack-plugin');
+const Dotenv = require('dotenv-webpack');
+const VueLoaderPlugin = require('vue-loader/lib/plugin');
+
 const buildPath = path.resolve(__dirname, 'public/');
 
 module.exports = {
-  mode: 'production',
+  mode: 'development',
```
Comment (AlexJump24) [inline webpack.prod.js:11]: Given this is the webpack production configuration, was it intended for this to be changed to development?

Comment (dannewnsjump24) [reply webpack.prod.js]: Good spot i've changed this back now, I did this to identify an issue in the build process that only showed up in development

File: resources/js/bootstrap.js [hunk 2]
```diff
@@ -50,9 +49,12 @@ orderFilters.init();
 const adminMenu = new AdminMenu();
 adminMenu.init();
 
-Vue.config.devtools = false;
+Vue.config.devtools = true;
```
Comment (AlexJump24) [inline resources/js/bootstrap.js:52]: This has been re-enabled, could we control this using `process.env.NODE_ENV`

Comment (dannewnsjump24) [reply resources/js/bootstrap.js]: Ive changed this now for this PR ill leave it hard coded ill look to bring in a change to use the node process for a few items then+

**Pull request merged**